### PR TITLE
add `.` to `zig-re-identifier` to highlight namespaced types

### PR DESCRIPTION
**Pull request by mega-dean**
This change adds `.` to the `zig-re-identifier` regex so that namespaced types are entirely highlighted using `font-lock-type-face`, instead of only highlighting the first namespace.

Before:
before

After:
after

I tried to look at what `zig-re-identifier` is used for to make sure there wouldn't be any unintended side effects from this change, but I can't tell for sure. I skimmed some zig files and it looked like it only affected struct field types and function arguments.

### CODE REVIEW

**Review by joachimschmidt557 — CHANGES_REQUESTED** [review]
Thanks for your contribution! I would like to suggest a change though: For me, `zig-re-identifier` should represent something similar to `IDENTIFIER` in the [zig grammar](https://ziglang.org/documentation/master/#Grammar), so `.` isn't allowed. As `.` do appear in _type names_ (`zig-re-type-annotation`) though, this is where I would add matching for `.`.

File: run_tests.sh [hunk 1]
```diff
@@ -19,4 +19,4 @@ INIT_PACKAGES="(progn \
       (package-install pkg))) \
   )"
 
-"${EMACS}" --eval "${INIT_PACKAGES}" --batch -l zig-mode.el -l tests.el -f ert-run-tests-batch-and-exit
+"${EMACS}" --eval "${INIT_PACKAGES}" --batch -l zig-mode.el -l test/zig-tests.el -f ert-run-tests-batch-and-exit
```
Comment (joachimschmidt557) [inline run_tests.sh:22]: Also, forgot to add this, this is a good catch, but would make more sense in a separate commit if possible :)

**Comment by mega-dean** [comment]
I update the PR so now it adds `(defconst zig-re-type "[[:word:]_.][[:word:]_.[:digit:]]*")` and uses that in `zig-re-type-annotation`.

At first I tried updating the `zig-re-type-annotation` regex to keep using `zig-re-identifier` and I got it working with this:
```
(defconst zig-re-type-annotation
  (concat (zig-re-grab zig-re-identifier)
          "[[:space:]]*:[[:space:]]*"
          zig-re-optionals-pointers-arrays
          (zig-re-grab (concat (zig-re-grab (concat zig-re-identifier "\\.?")) "*"))))
```
But I think it's probably better to just define the new constant. Let me know if you want me to change it though.

**Comment by joachimschmidt557** [comment]
Thanks! For now, this is perfectly fine. I think the regexes need to be revamped altogether in future to more closely match the zig grammar – or use tree sitter as basis.
